python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/monitor_sessions.py

```diff
@@ -50,42 +49,30 @@ def monitor_sessions(self: AvdStructuredConfigConnectedEndpoints) -> list | None
                         )
                         raise AristaAvdInvalidInputsError(msg)
 
-            monitor_session = {
-                "name": session_name,
-                "sources": [],
-                "destinations": [session._interface for session in session_configs_list if session.role == "destination"],
-            }
+            monitor_session = EosCliConfigGen.MonitorSessionsItem(
+                name=session_name,
+                sources=[],
+                destinations=[session._interface for session in session_configs_list if session.role == "destination"],
+            )
             source_sessions = [session for session in session_configs_list if session.role == "source"]
             for session in source_sessions:
-                source = {
-                    "name": session._interface,
-                    "direction": session.source_settings.direction,
-                }
-                if session.source_settings.access_group.name is not None:
-                    source["access_group"] = {
-                        "type": session.source_settings.access_group.type,
-                        "name": session.source_settings.access_group.name,
-                        "priority": session.source_settings.access_group.priority,
-                    }
-                append_if_not_duplicate(
-                    list_of_dicts=monitor_session["sources"],
-                    primary_key="name",
-                    new_dict=source,
-                    context="Monitor session defined under connected_endpoints",
-                    context_keys=["name"],
+                source = EosCliConfigGen.MonitorSessionsItem.SourcesItem(
+                    name=session._interface,
+                    direction=session.source_settings.direction,
                 )
+                if session.source_settings.access_group.name is not None:
+                    source.access_group = EosCliConfigGen.MonitorSessionsItem.SourcesItem.AccessGroup(
+                        type=session.source_settings.access_group.type,
+                        name=session.source_settings.access_group.name,
+                        priority=session.source_settings.access_group.priority,
+                    )
```

Comment:
This the properties have the same name and type you can do:
```suggestion
                    source.access_group = session.source_settings.access_group._cast_as(EosCliConfigGen.MonitorSessionsItem.SourcesItem.AccessGroup)
```